### ISSUE TITLE
[FEATURE] Install a package as a sole project

### ISSUE THREAD

**Issue by jfoclpf** [npm/cli#1578]
# What / Why
Imagine you have a big project with a lot of npm dependencies, but which is not supposed to have dependants, that is, a sole single project from which no other packages are supposed to depend on. An exemple is the npm package `autocosts`, 54 dependencies and 0 dependants.

When I install this package on a new machine or newly created directory, the file/directory structure becomes:

```
package-lock.json
node_modules/
 autocosts/
   node_modules/
```

## When
When I run `npm install <package>` on an empty directory or a new machine

### Current Behavior
It makes this folder/file structure with no `package.json` for the current package
```
package-lock.json
node_modules/
 autocosts/
   node_modules/
```

Then if I want to start the server `npm start` an error occurs, that is, the command `npm i autocosts && npm start` does not work

### Expected Behavior
The folder structure should just be with an extra option `npm i autocosts --sole`

```
(files of package)
package.json
package-lock.json
node_modules/
other_dirs/
```

Basically I'm requesting an extra option to make it possible to install everything directly in the current directory

**Comment by ljharb** [comment]
`npm start` is a development script - you're only supposed to run that when developing a project, which usually means you've git cloned it. `npm install` is for dependencies - something you run or require, but *not* something you run scripts in.

**Comment by ljharb** [comment]
iow, that particular package does not have correct instructions; "apps" (like that one) should not be installed in this way unless they provide a binary; the proper instructions would be to tell people to clone the repo.

**Comment by jfoclpf** [comment]
If I tell people to clone the repo with git clone, npm dependencies will
not be installed. We don't commit by default the content of node_modules/
directory.

**Comment by ljharb** [comment]
Step 1: clone the repo
Step 2: `npm install`
Step 3: `npm start`.

That's the way it's usually done as far as I'm aware.

**Comment by jfoclpf** [comment]
In that case I'd rather have
Step 1: `npm install package`
Step 2: `cd node_modules/package`
Step 3: `npm start`

Less programs to run. Furthermore in theory it's possible to have npm packages with no github projects attached

**Comment by ljharb** [comment]
Sure, it's possible, but it's a very unidiomatic way to use npm, and cding into node_modules - what should be a black box - is not a good practice.

**Comment by jfoclpf** [comment]
Wouldn't be simpler to have an extra option?

Step 1: `npm install package --{app|sole}`
Step 3: `npm start`

If the npm repository has already all the files I need, why do I need to fetch them in github?

**Comment by ljharb** [comment]
Because `npm start` is for an app - which needs dev deps. If you want a runnable binary, use "bin" and not "start".

**Comment by jfoclpf** [comment]
If I understand correctly, bin is for a CLI executable in the bin folder,
and in my case it is indeed a node app. But it's an app which needs to
start a server for it to work (basically it's a http server which serves a
website/calculator). And for the server to run in production I don't need
dev dependencies.

If all the files I need are in the npm repository, I still don't understand
why should I fetch them from github.

**Comment by jfoclpf** [comment]
> Step 1: clone the repo
> Step 2: `npm install`
> Step 3: `npm start`.
> 
> That's the way it's usually done as far as I'm aware.

Why should I fetch files from two different places?

**Comment by ljharb** [comment]
"an app which needs to start a server" works just fine as a binary.

**Comment by jfoclpf** [comment]
But how does it work exactly?
Where do I put the binary in my repo?
How do I generate the binary if I just have a node script?

**Comment by ljharb** [comment]
You put it anywhere you want, as long as "bin" in package.json points to it.

By "binary" i don't mean a binary file, i just mean an executable node JS file. So, whatever your node script does, that's what your "binary" would do.

**Comment by jfoclpf** [comment]
Hi @ljharb 
I was checking the [`bin` option](https://docs.npmjs.com/files/package.json#bin) in `package.json` and I still don't get how that can solve my problem. Such bin will be put in the folder `node_modules/.bin/` but when the user installs the package from scratch by doing `npm i autocosts`, he has no access to the bins. 

Therefore the question remains: how does a user install a sole npm project (i.e. with dependencies but no dependants) on a new machine or an empty directory, without the need to fetch the files from github since they are already in the npm repository?

**Comment by jfoclpf** [comment]
@ljharb I updated my request
https://github.com/npm/cli/issues/1578#issue-667864586

**Comment by jfoclpf** [comment]
@ljharb 
it's not the option I like the most, thus I requested this feature, but eventually I used your initial suggestion:
https://github.com/jfoclpf/autocosts#try-it-on-your-machine

Thanks anyway

**Comment by ljharb** [comment]
The user would invoke your package in either npx or an npm run-script.

**Comment by jfoclpf** [comment]
To run npm run-script you need package.json on the current directory, and
when you install a package on a newly created directory, a package.json is
not provided in said directory. Do kindly see my OP.

**Comment by darcyclarke** [comment]
npm `v6` is no longer in active development; We will continue to push security releases to `v6` at our team's discretion as-per our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy).

If your bug is preproducible on `v7`, please re-file this issue using our [new issue template](https://github.com/npm/cli/issues/new?assignees=&labels=Bug%2CNeeds+Triage%2CRelease+7.x&template=bug_7.yml&title=%5BBUG%5D+%3Ctitle%3E).

If your issue was a feature request, please consider opening a new [RRFC](https://github.com/npm/rfcs/issues) or [RFC](https://github.com/npm/rfcs). If your issue was a question or other idea that was not CLI-specific, consider opening a discussion on our [feedback repo](https://github.com/npm/feedback)

**Closing:** This is an automated message.
